### ISSUE TITLE
Filter server elements does not work correctly

### ISSUE THREAD

**Issue by j0ta29** [thim81/openapi-format#139]
YAML Input:
```
openapi: 3.0.1
servers:
  - url: 'https://server-a'
    x-stage-e-exclude: 
      - true
  - url: 'https://server-b'
    x-stage-e-exclude: 
      - true
  - url: 'https://server-c'
    x-stage-e-exclude: 
      - true
  - url: 'https://server-d'
    x-stage-e-exclude: 
      - true
  - url: 'https://server-e'
    x-stage-e-exclude: 
      - false
 ```
Filter-YAML:
```
flagValues:
  - x-stage-e-exclude: true
```

YAML Output:
```
openapi: 3.0.1
servers:
  - url: 'https://server-d'
    x-stage-e-exclude:
      - true
  - url: 'https://server-e'
    x-stage-e-exclude:
      - false
```

expected YAML Output:
```
openapi: 3.0.1
servers:
  - url: 'https://server-e'
    x-stage-e-exclude:
      - false

```

**Comment by thim81** [comment]
hi @j0ta29 

If the "flags" are just indicators to keep elements in or out; and you care less about the value, you could simply use "flags" like [this (see playground)](https://openapi-format-playground.vercel.app/?input=H4sIAAAAAAAAA53LPQ6AIAxA4Z1TdGMqaNx6G8T6kxAlFAzHl8SZhbzx5Xsi3y5eBIuZzKyE08tJSAEglBQI9JlzFLL2X%2Bh0ewAVJbuDsVV9KBsT5FS479ZB5wfdNui463YXhD%2FqBwStMAEAAA%3D%3D&config=H4sIAAAAAAAAA2WOMQrDMAxFrxI0xxfwGuhU6OA1i2gU11SxjSVDSundKxLoUjRJenr6b5DSFLy2TiM8iepUto2yCvgVWWy4JlZqgYyClTGKn%2FMwuGF3ohjJWe137gvNGcZDd7LWJLkcx7eqqWSZCjNWoeXnThKM%2F1%2BfcUrX2vWKOXZ7ZMoXbmzaivoIR2woLcWUkeHzBXOEkVvKAAAA).

I'll check why the "flagValues" are not filtering as expected.

**Comment by thim81** [comment]
hi @j0ta29 

We have a [PR](https://github.com/thim81/openapi-format/pull/140) ready that contains a fix for properly filtering flagValues for nested array values, like "servers".

If things progress, we will ship another version of openapi-format next week.

**Comment by thim81** [comment]
hi @j0ta29 

We just released openapi-format v1.23.1, which solves the reported bug.
You can see the result in the [playground](https://openapi-format-playground.vercel.app/?input=H4sIAAAAAAAAA6XLOw6AIBBF0Z5VTEcFaOxmNwjjJyFKGDAuX9TaWJDXvZuzR9psXBEG3eleMKWDEqMAUFBSQJBLzpHRmDcpK2sDOBVnO5OqO10onvC5b5ZToW8%2FNnrX6H2jp18%2F2cB0AfA3vyJYAQAA&config=H4sIAAAAAAAAA2WPMQrDMAxFr2I0JxfwGuhU6BDolEU0imuq2MaSIaX07lUT6FI0Sf%2FrfekFkquC19qogwdRGfK6UlIBvyCLDZfISnUkc8HCGK7IjcRPybnebb0oBuqtthu3mbz7oqYE3U4%2B1qyJcto5l6IxJxkyMxah%2BRcTZTT%2Fv3xclpuWpmdMoVmcIZ%2B4smEL6n3cP4BcY4gJGd4fA%2FSNYdUAAAA%3D)

If the result is as expected, feel free to close this item.

**Comment by j0ta29** [comment]
@thim81, thanks so much for the fast response and fix.
I can can confirm, that my problem is addressed and the issue can be closed.
But one note:
I played around on playground with another variant of my problem:

Input:
```
openapi: 3.0.1
servers:
  - url: 'https://server-a'
    x-stage:
      - a
  - url: 'https://server-b'
    x-stage:
      - b
  - url: 'https://server-c'
    x-stage:
      - c
  - url: 'https://server-d'
    x-stage:
      - d
  - url: 'https://server-e'
    x-stage:
      - e
``` 

Filter:
```
flagValues:
  - x-stage: e
```

Output:
```
openapi: 3.0.1
servers:
  - url: 'https://server-a'
    x-stage:
      - a
  - url: 'https://server-b'
    x-stage:
      - b
  - url: 'https://server-c'
    x-stage:
      - c
  - url: 'https://server-d'
    x-stage:
      - d
```

Perfect! But if I change the filter to the inverse
```
inverseFlagValues:
  - x-stage: e
```

Nothing is filtered:
```
openapi: 3.0.1
servers:
  - url: 'https://server-a'
    x-stage:
      - a
  - url: 'https://server-b'
    x-stage:
      - b
  - url: 'https://server-c'
    x-stage:
      - c
  - url: 'https://server-d'
    x-stage:
      - d
  - url: 'https://server-e'
    x-stage:
      - e
```

**Comment by thim81** [comment]
hi @j0ta29 

Good catch, I missed that case.
I created a new [PR](#141) to introduce the support for the filtering.

**Comment by thim81** [comment]
hi @j0ta29 

We just released 1.23.2 which contains the fix for the inverseFlagValues removal of nested array values.

See it in action in the [playground](https://openapi-format-playground.vercel.app/?input=H4sIAAAAAAAAA3XLOw6AIBCE4d5TbEcFaOz2Ngts1MQoATQe31c95eSfb8%2B6SV6YRte7oataTi2VOyJLR1mZzNxaruz9n6yYpxFdtjaZlL%2FxngWTAEjAJAISMUmAJEwUEL0BA6WmkhYBAAA%3D&config=H4sIAAAAAAAAA2WOsQrDMAxEf8VoTn7AayBToUOgUxZBFddUkY0ll5bSf69xoEvH053e3Rs0FQNvpdIAd6I8pX0nMQW%2FIWs7bpGNykItBVEeVJRmxnBBrqR%2BFedG9xzVMJB3tAoMnXk8NBF17oRztphEp8SMWen6K4i6tPy%2FfWxK1XK1E0qoraEhX7hzw2a029K3QyoxREGGzxfMIWXDzwAAAA%3D%3D)

Thanks for reporting this and helping to make openapi-format better, much appreciated.
